consensus/src/quorum_store/batch_store.rs

```diff
@@ -36,62 +35,70 @@ use std::{
 };
 use tokio::sync::oneshot;
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct PersistRequest {
-    pub digest: HashValue,
-    pub value: PersistedValue,
-}
-struct QuotaManager {
+pub type PersistRequest = PersistedValue;
+
+// Pub(crate) for testing only.
+pub(crate) struct QuotaManager {
     memory_balance: usize,
     db_balance: usize,
     batch_balance: usize,
     memory_quota: usize,
     db_quota: usize,
-    batch_quota: usize,
 }
 
 impl QuotaManager {
-    fn new(db_quota: usize, memory_quota: usize, batch_quota: usize) -> Self {
+    pub(crate) fn new(db_quota: usize, memory_quota: usize, batch_quota: usize) -> Self {
         assert!(db_quota >= memory_quota);
         Self {
             memory_balance: 0,
             db_balance: 0,
-            batch_balance: 0,
+            batch_balance: batch_quota,
             memory_quota,
             db_quota,
-            batch_quota,
         }
     }
 
     pub(crate) fn update_quota(&mut self, num_bytes: usize) -> anyhow::Result<StorageMode> {
-        if self.batch_balance + 1 > self.batch_quota {
+        if self.batch_balance == 0 {
```

Comment:
the name is very confusing with db_balance, one of them goes up, another goes down